pkg/api/websocket.go

```diff
@@ -203,6 +210,27 @@ func (a *StreamplaceAPI) HandleWebsocket(ctx context.Context) httprouter.Handle
 		}()
 
 		for {
+			r := msgLimiter.Reserve()
+			if !r.OK() {
+				log.Error(ctx, "rate limit exceeded, message rejected")
+
+				errorMsg := map[string]string{"error": "rate limit exceeded"}
+				errorBytes, _ := json.Marshal(errorMsg)
+				conn.WriteMessage(websocket.TextMessage, errorBytes)
+
+				continue
+			}
+
+			// wait for rate limit delay if there is one
+			delay := r.Delay()
+			if delay > 0 {
+				select {
+				case <-time.After(delay):
+				case <-ctx.Done():
+					return
+				}
+			}
+
```

Comment:
Not worried about this case - this is user --> streamplace messages, and zero of those exist right now - but we should have an IP-specific check for "number of websockets currently open" to handle that one person that's spamming us